x/poolmanager/create_pool.go

```diff
@@ -16,9 +16,6 @@ func (k Keeper) validateCreatedPool(ctx sdk.Context, poolId uint64, pool types.P
 	if pool.GetId() != poolId {
 		return types.IncorrectPoolIdError{ExpectedPoolId: poolId, ActualPoolId: pool.GetId()}
 	}
-	if !pool.GetAddress().Equals(types.NewPoolAddress(poolId)) {
-		return types.IncorrectPoolAddressError{ExpectedPoolAddress: types.NewPoolAddress(poolId).String(), ActualPoolAddress: pool.GetAddress().String()}
-	}
```

Comment:
Note tore viewer: this check is now invalid since we use contract address for cw pool

This seems like a defense-in-depth check which isn't useful anyways since address being correctly set should be tested by unit tests anyways